java/pattern/constraint/TypeConstraint.java

```diff
@@ -562,7 +577,24 @@ public boolean isPlays() {
 
         @Override
         public String toString() {
-            return PLAYS.toString() + SPACE + roleType;
+            StringBuilder syntax = new StringBuilder();
+
+            syntax.append(PLAYS).append(SPACE);
+            if (relationType!=null) {
+                if (!relationType.label().isPresent()) syntax.append(relationType);
+                else syntax.append(relationType.label().get().label());
+                syntax.append(COLON);
+            }
+
+            if (!roleType.label().isPresent()) syntax.append(roleType);
+            else syntax.append(roleType.label().get().label());
+
+            if (overriddenRoleType != null) {
+                syntax.append(SPACE).append(AS).append(SPACE);
+                if (!overriddenRoleType.label().isPresent()) syntax.append(overriddenRoleType);
+                else syntax.append(overriddenRoleType.label().get().label());
```

Comment:
Same as above: this logic is not needed given that `TypeVariable.toString()` already takes care of this.

Reply:
Fixed as above.